trustfall_core/src/ir/mod.rs

```diff
@@ -232,8 +220,7 @@ pub enum FoldSpecificFieldKind {
     Count, // Represents the number of elements in an IRFold's component.
 }
 
-static NON_NULL_INT_TYPE: Lazy<Type> =
-    Lazy::new(|| Type { base: BaseType::Named(Name::new("Int")), nullable: false });
+static NON_NULL_INT_TYPE: Lazy<Type> = Lazy::new(|| Type::new_named_type("Int", false));
```

Comment:
this and other common types like `String!` for `__typename` are probably best moved into the new `types` module I proposed